### ISSUE TITLE
Cannot change name of image on Android when testing

### ISSUE THREAD

**Issue by pleyland** [godotengine/godot#61022]
### Godot version

v3.4.4.stable.official [419e713a2]

### System information

Windows 10

### Issue description

 I used a lowercase letter instead of a capital when naming my image, and I ran the game on my phone. When I corrected the image file name the name does not change when testing the game again. I have tried uninstalling the app and clearing the data. I renamed the images again in godot, even renaming them to something completely different, running the app, and the changing them back to the desired name, and running it again, but that didn't help.
 
 I know that the image on my phone is not changing because I can reference the incorrectly named image from my code when the image is not present in my editor. I can continue but I cannot use the desired filename and I have to use the original filename that was an error.
 
 As you can imagine, if this always happens when renaming images during testing it could lead to catastrophic failures. 

### Steps to reproduce

Add image sprite with name with lowercase letter and reference uppercase in code. Export to phone. Change image name to uppercase. Export again and notice that the image reference does not work because the name has not changed

**Comment by allkhor** [comment]
@pleyland  Please upload a [minimal reproduction project](https://github.com/godotengine/godot/blob/master/CONTRIBUTING.md#provide-a-simple-example-project) to make this easier to troubleshoot.

**Comment by pleyland** [comment]
> @pleyland Please upload a [minimal reproduction project](https://github.com/godotengine/godot/blob/master/CONTRIBUTING.md#provide-a-simple-example-project) to make this easier to troubleshoot.

I did this, and I could not reproduce the issue, and when I tried my project again the bug no longer existed. This was after building it directly on my phone and seeing the bugs. I did not change anything about my project except create a print statement above the line of code that assigns the Sprite Texture name with a string that is composed of a variable string. After I printed the string and saw that it was correct, the project worked like magic. 

I think sometimes certain things are not 100% updated every time you compile the project unless you make edits.

Thanks for your help
